prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/rest/RestWorkFlowTask.java

```diff
@@ -0,0 +1,127 @@
+package com.redhat.parodos.tasks.rest;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflow.parameter.WorkParameter;
+import com.redhat.parodos.workflow.parameter.WorkParameterType;
+import com.redhat.parodos.workflow.task.BaseWorkFlowTask;
+import com.redhat.parodos.workflow.task.enums.WorkFlowTaskOutput;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestClientException;
+
+@Slf4j
+public class RestWorkFlowTask extends BaseWorkFlowTask {
+
+	private RestService restService;
+
+	public RestWorkFlowTask() {
+		restService = new RestServiceImpl();
+	}
+
+	RestWorkFlowTask(String beanName, RestService restService) {
+		this.restService = restService;
+		setBeanName(beanName);
+	}
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		LinkedList<WorkParameter> params = new LinkedList<>();
+		params.add(WorkParameter.builder().key("url").type(WorkParameterType.TEXT).optional(false)
+				.description("URL to send request to").build());
```

Comment:
not sure how it is in other places, but I'd extract the keys to constants or to a parameter class.
the need to repeat the exact key name and be aware of its type isn't great.

Reply:
I do not understand what u mean by 'repeat the key' and 'aware of its type'.
The key is defined with its type only once.
The user has to be aware of the type when the request is created.